src/test/groovy/org/gradle/android/TaskAvoidanceTest.groovy

```diff
@@ -0,0 +1,40 @@
+package org.gradle.android
+
+import org.gradle.util.GradleVersion
+import org.gradle.util.VersionNumber
+import spock.lang.Unroll
+
+class TaskAvoidanceTest extends AbstractTest {
+    @Unroll
+    def "Source Tasks are avoided with #gradleVersion and Android plugin #androidVersion"() {
+        assert gradleVersion instanceof GradleVersion
```

Comment:
After removing these, I learned that this helps with Intellij type inference. I added them back so that Intellij can properly show errors when these are passed to the functions incorrectly.

Reply:
Seems a bit contradictive to me to use a dyanmic language and then optimize the code for type inference by the IDE.

Reply:
Yeah makes sense. It was copied from here: https://github.com/gradle/android-cache-fix-gradle-plugin/blob/master/src/test/groovy/org/gradle/android/CrossVersionOutcomeAndRelocationTest.groovy#L20

Removed.